cpp/inc/bond/ext/grpc/unary_call.h

```diff
@@ -149,37 +164,64 @@ class unary_call final
     }
 
     /// @brief Get the server context for this call.
+    ///
+    /// @note May only be called on unary_call objects that haven't been
+    /// moved-from.
     const grpc::ServerContext& context() const
     {
         BOOST_ASSERT(_impl);
         return _impl->_context;
     }
 
     /// @brief Get the server context for this call.
+    ///
+    /// @note May only be called on unary_call objects that haven't been
+    /// moved-from.
     grpc::ServerContext& context()
     {
         BOOST_ASSERT(_impl);
         return _impl->_context;
     }
 
     /// @brief Get the request message for this call.
+    ///
+    /// @note May only be called on unary_call objects that haven't been
+    /// moved-from.
     const TRequest& request() const
     {
         BOOST_ASSERT(_impl);
         return _impl->_request;
     }
 
     /// @brief Get the request message for this call.
+    ///
+    /// @note May only be called on unary_call objects that haven't been
+    /// moved-from.
     TRequest& request()
     {
         BOOST_ASSERT(_impl);
         return _impl->_request;
     }
 
+    /// @brief Creates a std::shared_ptr<unary_call> from this instance.
+    ///
+    /// This instance may not be used after to_shared has been called; the
+    /// returned shared instance must be used instead.
+    ///
+    /// @note May only be called on unary_call objects that haven't been
+    /// moved-from.
+    std::shared_ptr<unary_call> to_shared()
+    {
+        return std::make_shared<unary_call>(std::move(*this));
+    }
+
     /// @brief Responds to the client with the given message and status.
     ///
     /// Only the first call to \p Finish or \p FinishWithError will be
     /// honored.
+    ///
+    /// @note May only be called on unary_call objects that haven't been
+    /// moved-from.
```

Comment:
I don't understand this comment. My understanding is that if you std::move the object to another call, like in your example (line 67), you later call Finish (line 53). The sentence has the opposite meaning for me.

Reply:
Is this clearer phrasing? "May not be called on this instance if it has been moved-from."

The two unary_call instances on lines 53 and 67 are different instances. When the instance from 67 is used to create the one that used on line 53, the one on 67 enters the "moved-from" state.